clients/socket-client.rst

```diff
@@ -1,2 +1,63 @@
 Socket Client
 =============
+
+The socket client uses the stream extension from PHP, which is integrated into
+the core.
+
+Features
+--------
+
+* TCP Socket Domain (``tcp://hostname:port``)
+* UNIX Socket Domain (``unix:///path/to/socket.sock``)
+* TLS / SSL encryption
+* Client Certificate (only for PHP > 5.6)
+
+Installation
+------------
+
+To install the Socket client, run:
+
+.. code-block:: bash
+
+    $ composer require php-http/socket-client
+
+Usage
+-----
+
+The Socket client need a :ref:`message factory <message-factory>` in order to
+to work::
+
+    use Http\Client\Socket\Client;
+
+    $options = [];
+    $client = new Client($messageFactory, $options);
+
+The available options are:
+
+:remote_socket: Specify the remote socket where the library should send the request to
+
+    * Can be a TCP remote: ``tcp://hostname:port``
+    * Can be a UNIX remote: ``unix://hostname:port``
+    * Do not use a TLS/SSL scheme, this is handle by the SSL option.
+    * If not set, the client will try to determine it from the request URI or host header.
```

Comment:
No, this option is where the client needs to be connected to send Http Requests, don't know how it works if we have a proxy. This is mainly due to support unix socket domain transport, like for docker.